Updated np_frontend_helpers.handle_where_and_array_bools() and tests to match the recent refactoring

hello-fri-end opened the pull request.

hello-fri-end commented:
Hi @Aarsh2001! Can you review this please?

Aarsh2001 commented:
Hi @hello-fri-end, can you please add the task description or link the comments. Thanks 🙂

Aarsh2001 reviewed (commented):
Please add the trello task link -:

hello-fri-end commented:
Hi @Aarsh2001! This PR is not linked to any open issue or Trello task. The goal here was to update ` np_frontend_helpers.handle_where_and_array_bools()` and all the tests that use this function to match the refactoring done by @CatB1t  in https://github.com/unifyai/ivy/pull/9692. The main thing to look at here is the updated function :


![latest-screenshot](https://user-images.githubusercontent.com/43880587/215406407-0dc5def0-00f8-4176-9c22-a3e95f1203f6.png)

And all the tests that use this function have been updated to match this change and the addition of `native_array` and `as_variable` flags to the `test_flags` object (https://github.com/unifyai/ivy/pull/9692). Here's an example:

![latest-screenshot](https://user-images.githubusercontent.com/43880587/215407259-2c854d0e-2d33-46cc-991f-3f5b85ca4392.png)

CatB1t commented on ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:
```diff
@@ -173,10 +173,9 @@ def test_frontend_function(*args, where=None, **kwargs):
 # noinspection PyShadowingNames
 def handle_where_and_array_bools(where, input_dtype, as_variable, native_array):
```
I'd suggest we take in the `test_flags` object as a parameter instead of `as_variable` and `native_array`, this will make the code cleaner and more compact

CatB1t commented on ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:
```diff
@@ -173,10 +173,9 @@ def test_frontend_function(*args, where=None, **kwargs):
-        return where, as_variable + [False], native_array + [False]
-    return where, as_variable, native_array
+        return where, input_dtype + ["bool"], as_variable + [False], native_array + [False],
+    return where, input_dtype, as_variable, native_array
```
Similarly, we would return a `test_flags` object.

CatB1t reviewed (changes requested):
Hello @hello-fri-end, thanks for the great work, just a small change and then we're good to merge!

hello-fri-end commented:
Hi @CatB1t ! I've updated the function and tests as you suggested. Also, I noticed that you missed ` helpers.test_frontend_method`  for ndarrays in your refactoring PR. For now, I have only updated the `handle_where_and_array_bools` function in that file. Let's merge this one and I can open a new PR to fix that too?

hello-fri-end commented:
> Hi @CatB1t ! I've updated the function and tests as you suggested. Also, I noticed that you missed ` helpers.test_frontend_method` for ndarrays in your refactoring PR. For now, I have only updated the `handle_where_and_array_bools` function in that file. Let's merge this one and I can open a new PR to fix that too?

Hey @CatB1t ! I have updated all the tests for `np.ndarray` that use the `np_frontend_helpers.handle_where_and_array_bools` to match the changes you made in https://github.com/unifyai/ivy/pull/10187. I also updated some tests that you had missed in your PR :)

CatB1t reviewed (approved):
@hello-fri-end LGTM! great work!
